modules/genomic_browser/php/endpoints/profiles.class.inc

```diff
@@ -0,0 +1,317 @@
+                    'select',
+                    'Date of Birth',
+                    array('maxlength' => 10)
+                ),
+                'External_ID'  => $this->_formElementObjectCreator(
+                    'External_ID',
+                    'text',
+                    'External ID',
+                    null
+                ),
+                'File'         => $this->_formElementObjectCreator(
+                    'File',
+                    'select',
+                    'Files',
+                    $any_options
+                ),
+                'SNP'          => $this->_formElementObjectCreator(
+                    'SNP',
+                    'select',
+                    'SNPs found',
+                    $any_options
+                ),
+                'CNV'          => $this->_formElementObjectCreator(
+                    'CNV',
+                    'select',
+                    'CNVs found',
+                    $any_options
+                ),
+                'CPG'          => $this->_formElementObjectCreator(
+                    'CPG',
+                    'select',
+                    'CPGs found',
+                    $any_options
+                ),
+               );
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
+            || !$user->hasPermission('genomic_browser_view_allsites')
+        ) {
+            $values['error'] = 'Forbidden';
+            return $values;
+        }
+        return $values;
+    }
+
+    /**
+     * Used to create (select, text, ...) element for form filter.
+     *
+     * @param string     $name   for element
+     * @param string     $type   for element
+     * @param string     $label  for element
+     * @param array|null $values for element
+     *
+     * @return object
+     */
+    private function _formElementObjectCreator(
+        string $name,
+        string $type,
+        string $label,
+        ?array $values
+    ) : object {
+        $element        = new \stdClass();
+        $element->name  = $name;
+        $element->type  = $type;
+        $element->label = $label;
+        if (isset($values)) {
+            $element->options = $values;
+        }
+        return $element;
+    }
+
+    /**
+     * An ETagCalculator provides the ability to calculate an ETag for
+     * an incoming HTTP request.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return string The value to use for the ETag header.
+     */
+    public function ETag(ServerRequestInterface $request): string
```

Comment:
Another function that probably belongs in the parent class.

Reply:
Hey @johnsaigle I think there's a reason to leave it here. @xlecours ?